event_sink_clickhouse/sinks/course_published.py

```diff
@@ -0,0 +1,233 @@
+"""
+Handler for the CMS COURSE_PUBLISHED event
+
+Does the following:
+- Pulls the course structure from modulestore
+- Serialize the xblocks and their parent/child relationships
+- Sends them to ClickHouse in CSV format
+
+Note that the serialization format does not include all fields as there may be things like
+LTI passwords and other secrets. We just take the fields necessary for reporting at this time.
+"""
+
+import csv
+import io
+import uuid
+
+import requests
+from django.utils import timezone
+
+from .base_sink import BaseSink
+
+
+class CoursePublishedSink(BaseSink):
+    """
+    Event sink for the COURSE_PUBLISHED signal
+    """
+    @staticmethod
+    def _get_detached_xblock_types():
+        """
+        Import and return DETACHED_XBLOCK_TYPES.
+        Placed here to avoid model import at startup and to facilitate mocking them in testing.
+        """
+        # pylint: disable=import-outside-toplevel,import-error
+        from xmodule.modulestore.store_utilities import DETACHED_XBLOCK_TYPES
+        return DETACHED_XBLOCK_TYPES
```

Comment:
This looks good! But I'm worried about compatibility issues with older releases. Do these imports work across the later releases?

Reply:
`DETACHED_XBLOCK_TYPES` has been in that location for 8 yrs so it should be good 👍

Reply:
Same for modulestore and courseoverview I guess.